src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -769,38 +747,110 @@ class ContentSidebar extends PureComponent<Props, State> {
         });
     };
 
+    /**
+     * Sorts and formats assignments, and then adds them to their task. 
+     *
+
+     * @param {TaskAssignment} data - Task create success callback
+     * @return {Task}
+     */
+    appendAssignmentsToTask(task: Task, assignments: Array<TaskAssignment>) {
+        if (!assignments) {
+            return task;
+        }
+
+        task.task_assignment_collection.entries = assignments.map((taskAssignment) => {
+            const { id, assigned_to, message, resolution_state } = taskAssignment;
+            return {
+                type: 'task_assignment',
+                id,
+                assigned_to,
+                message,
+                resolution_state
+            };
+        });
+        // Increment the assignment collection count by the number of new assignments
+        task.task_assignment_collection.total_count += assignments.length;
+        return task;
+    }
+
     /**
      * Adds a task to the tasks state and increases total_count.
      *
      * @param {Task} task - The newly created task from the API
+     * @param {SelectorItems} assignees - The list of users assigned to this task
+     * @param {Function} successCallback - Task create success callback
+     * @param {Function} errorCallback - Task create error callback
+
      * @return {void}
      */
-    createTaskSuccessCallback(task: Task): void {
-        const { tasks } = this.state;
-        if (tasks && tasks.entries) {
+    createTaskSuccessCallback(
+        task: Task,
+        assignees: SelectorItems,
+        successCallback: Function,
+        errorCallback: Function
+    ): void {
+        const { tasks, file } = this.state;
+
+        if (!file) {
+            throw getBadItemError();
+        }
+
+        if (!tasks || !tasks.entries) {
+            return;
+        }
+
+        const assignmentPromises = [];
+        assignees.forEach((assignee: SelectorItem) => {
+            // Create a promise for each assignment
+            const assignmentPromise = new Promise((resolve, reject) => {
+                this.api.getTaskAssignmentsAPI(false).createTaskAssignment({
+                    file,
+                    taskId: task.id,
+                    assignTo: { id: assignee.id },
+                    successCallback: (data: TaskAssignment) => resolve(data),
+                    errorCallback: (e) => {
```

Comment:
Maybe move this into it's own method

Reply:
Feels kind of weird to have to pass the reject call, unless you're talking about just moving 3 lines?

Reply:
I think @pramodsum  maybe meant create something like ```this.createTaskAssignment``` which returns a promise. Seems like a good idea for testability

Reply:
gotcha, I can do that!

Reply:
^ Yup that's what i meant :)